on-target-tests/tests/i2c_tests/non_blocking.rs

```diff
@@ -359,3 +359,34 @@ pub async fn transaction<A: ValidAddress>(
     .collect();
     assert_eq!(g, h);
 }
+
+pub async fn transaction_iter<A: ValidAddress>(state: &mut State, addr: A) {
+    use i2c_write_iter::non_blocking::I2cIter;
+    reset(state, addr);
+
+    let samples: FIFOBuffer = Generator::seq().take(25).collect();
+    let controller = state.controller.as_mut().expect("controller's missing.");
+    let case = async {
+        controller
+            .transaction_iter(
+                addr,
+                [i2c_write_iter::Operation::WriteIter(
+                    samples.clone().into_iter(),
```

Comment:
Perhaps this should be:
```suggestion
                    samples.iter().cloned(),
```
Advantages:
- doesn't clone the whole buffer
- works the same way if samples is a slice